FTB with json-c 0.19

**Newsboat version (copy the output of `newsboat -v` or the first line of `git show`)**: https://github.com/newsboat/newsboat/pull/3402 https://github.com/newsboat/newsboat/pull/3403

```
In file included from include/newsblurapi.h:4:
In file included from /opt/homebrew/Cellar/json-c/0.19/include/json-c/json.h:33:
/opt/homebrew/Cellar/json-c/0.19/include/json-c/linkhash.h:360:36: error: unused function 'lh_table_head' [-Werror,-Wunused-function]
static _LH_INLINE struct lh_entry *lh_table_head(const lh_table *t)
                                   ^
/opt/homebrew/Cellar/json-c/0.19/include/json-c/linkhash.h:376:33: error: unused function 'lh_get_hash' [-Werror,-Wunused-function]
static _LH_INLINE unsigned long lh_get_hash(const struct lh_table *t, const void *k)
                                ^
/opt/homebrew/Cellar/json-c/0.19/include/json-c/linkhash.h:398:25: error: unused function 'lh_entry_k' [-Werror,-Wunused-function]
static _LH_INLINE void *lh_entry_k(const struct lh_entry *e)
                        ^
/opt/homebrew/Cellar/json-c/0.19/include/json-c/linkhash.h:408:23: error: unused function 'lh_entry_k_is_constant' [-Werror,-Wunused-function]
static _LH_INLINE int lh_entry_k_is_constant(const struct lh_entry *e)
                      ^
/opt/homebrew/Cellar/json-c/0.19/include/json-c/linkhash.h:419:25: error: unused function 'lh_entry_v' [-Werror,-Wunused-function]
static _LH_INLINE void *lh_entry_v(const struct lh_entry *e)
                        ^
/opt/homebrew/Cellar/json-c/0.19/include/json-c/linkhash.h:428:24: error: unused function 'lh_entry_set_val' [-Werror,-Wunused-function]
static _LH_INLINE void lh_entry_set_val(struct lh_entry *e, void *newval)
                       ^
/opt/homebrew/Cellar/json-c/0.19/include/json-c/linkhash.h:438:36: error: unused function 'lh_entry_next' [-Werror,-Wunused-function]
static _LH_INLINE struct lh_entry *lh_entry_next(const struct lh_entry *e)
                                   ^
/opt/homebrew/Cellar/json-c/0.19/include/json-c/linkhash.h:448:36: error: unused function 'lh_entry_prev' [-Werror,-Wunused-function]
static _LH_INLINE struct lh_entry *lh_entry_prev(const struct lh_entry *e)
                                   ^
8 errors generated.
```

--- a/Makefile
+++ b/Makefile
@@ -22,7 +22,9 @@ CXX_FOR_BUILD?=$(CXX)
 # compiler and linker flags
 DEFINES=-DLOCALEDIR='"$(localedir)"' -DCATCH_AMALGAMATED_CUSTOM_MAIN
 
-WARNFLAGS=-Werror -Wall -Wextra -Wunreachable-code
+# `-Wno-unused-function` specified to work around json-c 0.19
+# https://github.com/json-c/json-c/issues/941
+WARNFLAGS=-Werror -Wall -Wextra -Wunreachable-code -Wno-unused-function
 INCLUDES=-Iinclude -Istfl -Ifilter -I. -Irss -I$(CARGO_TARGET_DIR)/cxxbridge/
 # Keep in sync with c++ version specified in FFI build.rs
 BARE_CXXFLAGS=-std=c++17 -O2 -ggdb $(INCLUDES)

